### DISCUSSION TITLE
co-location after the fact

### DISCUSSION THREAD

**Discussion by sullyj3** [jj-vcs/jj#2230] [Q&A] [answered]
If I forget to init with `--git-repo=.`, or clone with `--colocate`, is there a way to convert to a colocated repo? I tested just moving `.jj/repo/store/git` to `.git` in a duplicate of my repo but it broke.

**Comment by ilyagr** [comment, marked as answer]
I would try `jj git clone --colocate repo/.jj/store/git new_repo`. Or just `git clone` and then `jj init --git-repo=.`. This will lose all of your change ids, you might as well clone from the remote.

**Comment by martinvonz** [comment]
You can move `.jj/repo/store/git` to `.git` and also run `echo -n ../../../.git > .jj/repo/store/git_target` to tell the git commit backend where to find the git repo.

**Reply by sullyj3** [reply]
After doing this, `jj` commands seem to be working ok, `git log` works, but `git status` gives `fatal: this operation must be run in a work tree`.

**Reply by yuja** [reply]
You'll probably need to fix up .git/config to turn off the bare flag.
https://stackoverflow.com/questions/10637378/how-do-i-convert-a-bare-git-repository-into-a-normal-one-in-place

**Reply by sullyj3** [reply]
Comparing with another normal git repo, there seem to be some missing files. my-project is the one I'm trying to convert to colocated:
```
⮞ diff (ls -1 dev/jj/.git | psub) (ls -1 dev/my-project/.git | psub)
1d0
< branches
7d5
< index
9d6
< logs
11d7
< packed-refs
```

**Reply by sullyj3** [reply]
Ahh, so it's bare. Yup I think just re-cloning is the easiest method for now.

**Reply by Zambito1** [reply]
The documentation says to provide feedback when doing this, so I will do that here. I ran this while I was on an empty change on my `dev` branch, and I ended up on a new change on `master`, with all of the changes added to `dev` since the last merge. It took me a second to orient myself with `jj op log` and `jj log --at-operation` to figure out where I was before doing this. All I had to do was `jj rebase -r @ -d dev` and then I was good to go, but it was confusing for a moment.

**Reply by martinvonz** [reply]
I suspect it might have helped to run `jj git export` before the other steps there. Once we add a command for this operation, we should make sure to include that step. Thanks for the report.

**Comment by necauqua** [comment]
I think this operation (colocation after the fact) should be a jj command, now with colocation being promoted as very usable and with this being brought up more than once.

At the very least the steps to do it manually should be in the docs somewhere.

**Comment by andrewaylett** [comment]
Having just gone through the steps, with an empty working commit:

```
jj git export
mv .jj/repo/store/git ./.git
echo -n ../../../.git > .jj/repo/store/git_target
git config --local --bool core.bare false
jj new @-
```

This left me with a (different) empty working commit, and `git status` having the same view of the working copy as `jj status`.

**Reply by yuja** [reply]
Can you try the documented steps, which is slightly updated? I have no idea if that matters, though.
https://jj-vcs.github.io/jj/latest/git-compatibility/#converting-a-repo-into-a-co-located-repo
